src/Enum.php

```diff
@@ -28,6 +28,13 @@ abstract class Enum implements \JsonSerializable
      */
     protected $value;
 
+    /**
+     * Enum key, the constant name
+     *
+     * @var string
+     */
+    protected $key;
```

Comment:
I would have liked to have the property as private instead of protected but this doesn't seems to be the pattern here.

Reply:
👍  let's make it private, it's ok if it's not 100% consistent, we can always change from private to protected if there's a reason.

Reply:
done